kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/WorkloadBenchmark.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults.benchmarks;
+
+import static io.delta.kernel.defaults.benchmarks.BenchmarkUtils.*;
+
+import io.delta.kernel.defaults.engine.DefaultEngine;
+import io.delta.kernel.engine.Engine;
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import org.apache.hadoop.conf.Configuration;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.results.format.ResultFormatType;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.TimeValue;
+
+/**
+ * Generic JMH benchmark for all workload types. Automatically loads and runs benchmarks based on
+ * JSON workload specifications.
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(value = 1, warmups = 1)
+@Warmup(iterations = 3, time = 1)
+@Measurement(iterations = 5, time = 1)
+public class WorkloadBenchmark {
+
+  /**
+   * Dynamic benchmark state that reads the workload specification and sets up the workload runner
+   */
+  @State(Scope.Thread)
+  public static class BenchmarkState {
+    /**
+     * The json representation of the workload specification. Note: This parameter will be set
+     * dynamically by JMH. The value is set in the main method.
+     */
+    @Param({})
+    private String workloadSpecJson;
+
+    /**
+     * The base directory containing workload tables. This is used to resolve relative table paths.
+     * Note: This parameter will be set dynamically by JMH. The value is set in the main method.
+     */
+    @Param({})
+    private String baseWorkloadDir;
+
+    /** The workload runner initialized for this benchmark invocation. */
+    private WorkloadRunner runner;
+
+    /**
+     * Setup method that runs before each benchmark invocation. Reads the workload specification
+     * from the given path and initializes the corresponding workload runner.
+     *
+     * @throws Exception If any error occurs during setup.
+     */
+    @Setup(Level.Invocation)
+    public void setup() throws Exception {
```

Comment:
Is it possible for WorkloadBenchmark to be abstract? and it has a single abstract method `createEngine`?

So someone can easily have:

```java
class WorkloadBenchmarkWithDefaultEngine extends AbstractWorkloadBenchmark {
  @Override
  public Engine createEngine() {
    return DefaultEngine....
  }
}

class WorkloadBenchmarkWithDeltaSparkDSv2Engine extends AbstractWorkloadBenchmark {
  @Override
  public Engine createEngine() {
    return CustomDeltaSparkDSv2Engine..
  }
}
```